### ISSUE TITLE
feature needed: writeMetadata for handle::file

### ISSUE THREAD

**Issue by weilewei** [constantinpape/z5#113]
In file https://github.com/constantinpape/z5/blob/master/include/z5/metadata.hxx, there is no writeMetadata for 
```c
inline void writeMetadata(const handle::File & handle, const DatasetMetadata & metadata)
```
I ran into issue that 
once I call `createFile` from `z5/file.hxx`, it complains that 
`error: no matching function for call to 'writeMetadata(z5::handle::File&, z5::Metadata&)'
`

**Comment by weilewei** [comment]
I wrote several lines of code for metadata.hxx, not sure if that helps:

```cpp
    inline void writeMetadata(const handle::File & handle, const Metadata & metadata) {
        auto filePath = handle.path();
        filePath /= ".zgroup";
        nlohmann::json j;
        j["zarr_format"] = metadata.zarrFormat;
        fs::ofstream file(filePath);
        file << std::setw(4) << j << std::endl;
        file.close();
    }
```
Let me know if this looks good to you. Or I can create a pull request for your reference. My current project that I am working on this summer requires creating a file. Or let me know if I am going to a wrong direction. Thanks.

**Comment by constantinpape** [comment]
Thanks for noticing this.
Would be good to add `writeMetadata` for the file handle as well.
What you proposed looks mostly fine, however this will only work for zarr files (same issue is in the version for group).
Please change the line 
```
filePath /= ".zgroup"';
```
to 
```
filePath = metadata.isZarr ? ".zgroup" : "attributes.json";
```
If you make a PR for this, could you fix the issue in https://github.com/constantinpape/z5/blob/master/include/z5/metadata.hxx#L266 too?
Thanks a lot.

**Comment by weilewei** [comment]
Thanks for your suggestion. Could you please provide the permission of this repos for me, so I can push my branch to your repo and then create a pull request?

Also, I noticed that for writing file, is it better to write in this way:
```
filePath = metadata.isZarr ? ".zfile" : "attributes.json";
``` 
instead of "zgroup" in this line?

Let me know if you have any suggestion. Thanks.

**Comment by constantinpape** [comment]
> Could you please provide the permission of this repos for me, so I can push my branch to your repo and then create a pull request?

No need for that, you can just fork the repo and make a PR from your fork.

> filePath = metadata.isZarr ? ".zfile" : "attributes.json"

No, the zarr spec does not make a real difference between filesa and group, so you need to keep `.zgroup`.

**Comment by weilewei** [comment]
Good found! Thanks for your suggestion! Sorry for any confusion I made before. Please see the PR #114. Let me know if you have any concerns.

**Comment by constantinpape** [comment]
> Sorry for any confusion I made before

No worries.

> Please see the PR #114. Let me know if you have any concerns.

There is an issue with some operatores, otherwise it looks good. I added a review about it; once you fix these issues it should be good to go.

**Comment by constantinpape** [comment]
Fixed by #114
